theunderground/admin.py

```diff
@@ -0,0 +1,122 @@
+from werkzeug.security import generate_password_hash, check_password_hash
+from flask import render_template, url_for, flash, redirect
+from room import app, db
+from models import ConciergeMii, User
+from flask_login import login_required
+from forms import LoginForm,KillMii,ConciergeForm
+import datetime
+import shutil
+from flask_login import current_user, login_user
+descrutive_warning = '''
+  _____   ____ _______ ______ _   _ _______ _____          _      _  __     __
+ |  __ \ / __ \__   __|  ____| \ | |__   __|_   _|   /\   | |    | | \ \   / /
+ | |__) | |  | | | |  | |__  |  \| |  | |    | |    /  \  | |    | |  \ \_/ / 
+ |  ___/| |  | | | |  |  __| | . ` |  | |    | |   / /\ \ | |    | |   \   /  
+ | |    | |__| | | |  | |____| |\  |  | |   _| |_ / ____ \| |____| |____| |   
+ |_|___  \____/  |_|__|______|_|_\_| _|_| _|_____/_/____\_\______|______|_|   
+ |  __ \|  ____|/ ____|/ ____|  __ \| |  | |__   __|_   _\ \    / /  ____|    
+ | |  | | |__  | (___ | |    | |__) | |  | |  | |    | |  \ \  / /| |__       
+ | |  | |  __|  \___ \| |    |  _  /| |  | |  | |    | |   \ \/ / |  __|      
+ | |__| | |____ ____) | |____| | \ \| |__| |  | |   _| |_   \  /  | |____     
+ |_____/|______|_____/ \_____|_|  \_\\____/   |_|  |_____|   \/   |______|
+ '''
+enabled = '''
+             _           _                          _     _          _   
+           | |         (_)                        | |   | |        | |  
+   __ _  __| |_ __ ___  _ _ __     ___ _ __   __ _| |__ | | ___  __| |  
+  / _` |/ _` | '_ ` _ \| | '_ \   / _ \ '_ \ / _` | '_ \| |/ _ \/ _` |  
+ | (_| | (_| | | | | | | | | | | |  __/ | | | (_| | |_) | |  __/ (_| |_ 
+  \__,_|\__,_|_| |_| |_|_|_| |_|  \___|_| |_|\__,_|_.__/|_|\___|\__,_(_)
+'''
+disabled = '''
+            _           _             _ _           _     _          _ 
+           | |         (_)           | (_)         | |   | |        | |
+   __ _  __| |_ __ ___  _ _ __     __| |_ ___  __ _| |__ | | ___  __| |
+  / _` |/ _` | '_ ` _ \| | '_ \   / _` | / __|/ _` | '_ \| |/ _ \/ _` |
+ | (_| | (_| | | | | | | | | | | | (_| | \__ \ (_| | |_) | |  __/ (_| |
+  \__,_|\__,_|_| |_| |_|_|_| |_|  \__,_|_|___/\__,_|_.__/|_|\___|\__,_|
+'''
+print('You have entered the underground')
+print('!! WARNING !!')
+print('Changes made at /theunderground/admin with a login are potentially destructive')
+#print(descrutive_warning)
+print('For security, the underground is **disabled** by default')
+print('In order to activate it, please enter the **password** set by the admin into the variable below (line 20)')
+pw = '123'
+#enabled = True
+hash = 'pbkdf2:sha256:150000$isdrNS5h$ee21ae24974917651ba34d249bce5c1f4274f81625b3d4d20ce0cfe9f3293133'
```

Comment:
I think this should be loaded from the config, if at all.. not quite sure what it is.

Reply:
I made a new password system zurgeg, over in the merch-poc branch, the password.py file. It generates a very secure SHA256 hash. :)